packages/core/src/IntersectionDetector/IntersectionDetector.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useRef } from 'react'
+import { PropTypes } from '@dhis2/prop-types'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+/**
+ * @module
+ *
+ * @param {IntersectionDetector.PropTypes} props
+ * @returns {React.Component}
+ */
+export const IntersectionDetector = ({
+    threshold,
+    onVisibilityChange,
+    children,
+    className,
+    rootRef,
+}) => {
+    // use ref to prevent unnecessary re-render
+
+    // @var {Object}
+    // @prop {IntersectionObserver} current
+    const observer = useRef()
+
+    // @var {Object}
+    // @prop {bool} current
+    const isIntersecting = useRef()
+
+    // @var {Object}
+    // @prop {HTMLElement} current
+    const intersectionRef = useRef()
+
+    useEffect(() => {
+        const rootEl = rootRef.current
+        const intersectionEl = intersectionRef.current
+
+        if (rootEl && intersectionEl && !observer.current) {
+            const onIntersection = entries => {
+                // Currently there's no way to supply multiple thresholds,
+                // so a single entry can be assumed safely
+                const [entry] = entries
+
+                // Make sure the callback is not called multilpe times
+                // if there is no change
+                const intersectionChange =
+                    entry.isIntersecting !== isIntersecting.current
+
+                if (intersectionChange) {
+                    isIntersecting.current = entry.isIntersecting
+                    onVisibilityChange({ visible: entry.isIntersecting })
+                }
+            }
+
+            const observerOptions = { root: rootEl, threshold }
+            const intersectionObserver = new IntersectionObserver(
```

Comment:
For `ResizeObserver` we decided to add a polyfill. Perhaps we should do the same here. Support seems similar (to be honest, support seems pretty decent for both these `*Observer` APIs, so I'm not 100% sure why we decided to add the polyfill for the `ResizeObserver` in the first place... Quite possibly because the specifications for the API are still in draft status 🤷 )

Reply:
It's supported in all browsers that we support, wondering why you added a polyfill.
I'd prefer not to add one as that's ultimately the responsibility of the app

Reply:
> It's supported in all browsers that we support, wondering why you added a polyfill.
I'd prefer not to add one as that's ultimately the responsibility of the app

I second this, and maybe we should remove the resizeobserver polyfill. I think we should also add https://github.com/amilajack/eslint-plugin-compat and lint browser support for the APIs we use. We should set violations to errors so that we can catch usage of unsupported APIs.

Reply:
> I second this, and maybe we should remove the resizeobserver polyfill.
👍